atlp-rwanda/knights-ecomm-fe
# #48 implement buyer order tracking

#### What does this PR do?
This PR adds the order tracking feature for buyers. It shows the current status of each order and allows buyers to view all their orders and update the status of specific orders. 

#### Description of Task to be completed?
Task Involved:
- Display a list of all orders on the orders page.
- Allow updating the status of a specific order.
- Manage the state of all orders using Redux.
- Style the orders list using Tailwind CSS.
- Ensure the UI is responsive and looks good on desktop, tablet, and mobile devices.
- Show a loading indicator while data is being fetched.
- Implement an "Order Not Found" page for invalid order IDs.
- Write unit tests for the Redux slice and the React components to ensure proper functionality.

#### How should this be manually tested?
- Ensure You have backend service running and set VITE_APP_API_URL in your .env file.
- Clone this repository and navigate to the project directory
  ```bash
  git clone -b ft-register https://github.com/atlp-rwanda/knights-ecomm-fe.git
  cd knights-ecomm-fe
  git checkout ft-buyer-orders
  npm i
  npm run dev
  ```
#### What are the relevant pivotal tracker/Trello stories?
#48 
#### Screenshots 
![image](https://github.com/atlp-rwanda/knights-ecomm-fe/assets/82057476/fd1448b7-f7ec-4435-aba1-bef3bc8365dd)
![image](https://github.com/atlp-rwanda/knights-ecomm-fe/assets/82057476/968fc3fe-a4d4-451b-8c25-0bd17038bb74)
![image](https://github.com/atlp-rwanda/knights-ecomm-fe/assets/82057476/098440aa-6be4-4624-a633-60ee9e3f0bb6)
